nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/test/java/org/nuxeo/ecm/platform/audit/AbstractAuditStorageTest.java

```diff
@@ -167,4 +171,20 @@ public void testIn() throws Exception {
         assertEquals(NUM_OF_EVENTS, list.size());
     }
 
+    @Test
+    public void testQueryWithDate() {
+        // check that a query with a date doesn't raise any error
+        AuditQueryBuilder dateBuilder = new AuditQueryBuilder();
+        dateBuilder.predicates(Predicates.gte(LOG_EVENT_DATE, Calendar.getInstance()));
```

Comment:
Your code fix tests for `ZonedDateTime` but your test passes in a `Calendar`.  Is there a conversion taking place I am not seeing?  Is the new code that checks for a `ZonedDateTime` being exercised by the test?

Reply:
Yes between this layer and the one with the bug there's a conversion going on (`Literals.toLiteral`). And yes the code is exercised, it fails without the fix.